templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -340,84 +316,115 @@ contents:
         fi
       fi
 
-      # wait for DHCP to finish, verify connection is up
-      counter=0
-      configured=false
-
-      while [ $counter -lt 5 ]; do
-        sleep 5
-        # check if connection is active
-        if nmcli --fields GENERAL.STATE conn show "$ovs_interface" | grep -i "activated"; then
-          echo "OVS successfully configured"
-          update_nm_conn_files "$bridge_name" "$port_name"
-          copy_nm_conn_files
-          ip a show "$bridge_name"
-          configure_driver_options ${iface}
-          configured=true
-          break
-        fi
-        counter=$((counter+1))
-      done
-      if [ $configured != true ]; then
-        echo "WARN: OVS did not succesfully activate NM connection. Attempting to bring up connections"
-        counter=0
-        while [ $counter -lt 5 ]; do
-          if nmcli conn up "$ovs_interface"; then
-            echo "OVS successfully configured"
-            update_nm_conn_files "$bridge_name" "$port_name"
-            copy_nm_conn_files
-            configure_driver_options ${iface}
-            configured=true
-            break
-          fi
-          sleep 5
-          counter=$((counter+1))
-        done
-      fi
-      if [ $configured != true ]; then
-        echo "ERROR: Failed to activate $ovs_interface NM connection"
-        exit 1
-      fi
+      configure_driver_options "${iface}"
+      update_nm_conn_files "$bridge_name" "$port_name"
+      persist_nm_conn_files
     }
 
+    # Used to remove a bridge
     remove_ovn_bridges() {
       bridge_name=${1}
       port_name=${2}
-      iface=""
-      if nmcli connection show ovs-port-${port_name} &> /dev/null; then
-        iface=$(nmcli --get-values connection.interface-name connection show ovs-port-${port_name})
-      fi
 
       # Reload configuration, after reload the preferred connection profile
       # should be auto-activated
       update_nm_conn_files ${bridge_name} ${port_name}
       rm_nm_conn_files
-      nmcli c reload
-      sleep 5
 
       # NetworkManager will not remove ${bridge_name} if it has the patch port created by ovn-kubernetes
       # so remove explicitly
       ovs-vsctl --timeout=30 --if-exists del-br ${bridge_name}
-
-      # In some cases the preferred connection profile is not auto-activated
-      # (maybe due to differences in NM versions) so try to activate it
-      # explicitly.
-      if [ -n "$iface" ]; then
-        nmcli device connect $iface
-      fi
     }
 
-    # removes any previous ovs configuration
+    # Removes any previous ovs configuration
     rollback() {
+      echo "Reverting any previous OVS configuration"
```

Comment:
should be part of the first commit?

Reply:
Yes, I actually wanted to squash them but I will leave it as is for now and move relevant things to the first commits. Same answer for similar comments.

Reply:
I changed the order of commits, made more sense, and the second just adds the rollback on boot.